src/node_quic_socket.cc

```diff
@@ -134,21 +134,109 @@ void JSQuicSocketListener::OnServerBusy(bool busy) {
   Socket()->MakeCallback(env->quic_on_socket_server_busy_function(), 1, &arg);
 }
 
-void JSQuicSocketListener::OnDone() {
+void JSQuicSocketListener::OnEndpointDone(QuicEndpoint* endpoint) {
   Environment* env = Socket()->env();
   HandleScope scope(env->isolate());
   Context::Scope context_scope(env->context());
-  Socket()->MakeCallback(env->ondone_string(), 0, nullptr);
+  MakeCallback(
+      env->isolate(),
+      endpoint->object(),
+      env->ondone_string(),
+      0, nullptr);
 }
 
 void JSQuicSocketListener::OnDestroy() {
   // Do nothing here.
 }
 
+QuicEndpoint::QuicEndpoint(
+    Environment* env,
+    Local<Object> wrap,
+    QuicSocket* listener,
+    Local<Object> udp_wrap) :
+    BaseObject(env, wrap),
```

Comment:
mini style nit: `:` goes on the line of the first initializer

Reply:
I use this pattern fairly extensively throughout. These can be updates later on in a separate PR